### PR TITLE
FIX - Putting brain with another organ manipulation surgery

### PR DESCRIPTION
**Pull request by RengaN02**
## About The Pull Request

Even if the head is closed in a body without a brain, we could start organ manipulation surgery anywhere in the body and then select head from the bottom right and insert a brain. In other words, we can putting brain to head without head organ manipulation surgery, but this fix allows not happen anymore.

## Why It's Good For The Game

Its a bugfix

## Changelog

:cl:
fix: Putting brain to body in another zone organ manip surgery
/:cl:

### CODE REVIEW

**Comment by Cadunkus** [comment]
Wait so this allows you to remove the brain of someone with H.A.R.S.? Is that it or?

**Comment by RengaN02** [comment]
> Wait so this allows you to remove the brain of someone with H.A.R.S.? Is that it or?

If you wanna remove brain of someone with H.A.R.S you can use organ manip in chest, this fix will not affect it

**Comment by RengaN02** [comment]
> > Wait so this allows you to remove the brain of someone with H.A.R.S.? Is that it or?
> 
> If you wanna remove brain of someone with H.A.R.S you can use organ manip in chest, this fix will not affect it

Oh i think this fix not work in someone with H.A.R.S but i can fix it.

File: code/modules/surgery/surgery_helpers.dm [hunk 1]
```diff
@@ -11,6 +11,17 @@
 	else
 		return 0.5
 
+/proc/get_organmanip_location_isopened(mob/living/carbon/located_mob, location)
```
Comment (san7890) [inline code/modules/surgery/surgery_helpers.dm:14]: this needs a better name and some dmdocing https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md

File: code/modules/surgery/surgery_helpers.dm [hunk 2]
```diff
@@ -11,6 +11,17 @@
...
+	for(var/datum/surgery/procedure in located_mob.surgeries)
```
Comment (san7890) [inline code/modules/surgery/surgery_helpers.dm:15]: if we are guaranteed to not have any nulls or things not of the `/datum/surgery` type, use `as anything` like this

```suggestion
	for(var/datum/surgery/procedure as anything in located_mob.surgeries)
```

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/surgery/surgery_helpers.dm (hunk 1 above)
Comment (RengaN02) [reply code/modules/surgery/surgery_helpers.dm:14]: How is this: 
/proc/isLocationOpened

**Review by strandsofivy — COMMENTED** [review]
This PR doesn't actually touch surgery code, is it really bugged in the first place?

For reference, the code below is for surgery starting and it implements location checks correctly.

https://github.com/tgstation/tgstation/blob/92957faff56579619c4f2f3a4dad8311f7c542a8/code/modules/surgery/surgery.dm#L106-L124

Instead, the code that's actually being edited looks to be pretty ancient surgery functionality that we realistically don't need anymore because we have organ manipulation surgeries.

**Comment by RengaN02** [comment]
> Instead, the code that's actually being edited looks to be pretty ancient surgery functionality that we realistically don't need anymore because we have organ manipulation surgeries.

Yes, it has no effect on organ manipulation surgery, but it fixes the bug that occurs when putting the brain. Think about it, you have skin on your face, but someone skips the skin and puts something in your head

**Comment by distributivgesetz** [comment]
> Yes, it has no effect on organ manipulation surgery, but it fixes the bug that occurs when putting the brain. Think about it, you have skin on your face, but someone skips the skin and puts something in your head

I was wondering if we need the code in `attack` around because this is two implementations of the same mechanic, brain insertion.

Currently on master we have the ability to place a brain inside of a head at any time with the brain's `attack` implementation. With this PR you can only put the brain in if we have an organ maniplation surgery that's at the insert organ step. This is already handled by organ manipulation surgery, which makes the code here completely redundant, since surgery would cancel the `attack()` chain anyways.

**Comment by RengaN02** [comment]
> > Yes, it has no effect on organ manipulation surgery, but it fixes the bug that occurs when putting the brain. Think about it, you have skin on your face, but someone skips the skin and puts something in your head
> 
> I was wondering if we need the code in `attack` around because this is two implementations of the same mechanic, brain insertion.
> 
> Currently on master we have the ability to place a brain inside of a head at any time with the brain's `attack` implementation. With this PR you can only put the brain in if we have an organ maniplation surgery that's at the insert organ step. This is already handled by organ manipulation surgery, which makes the code here completely redundant, since surgery would cancel the `attack()` chain anyways.

attack = click

Everything you click is counted as an attack in the code, and what we do here is that when you click with a brain to a human, it places it on the human. What I have fixed here is to prevent the person from placing his head even if his head is closed when he clicks on a person without a brain.

**Comment by distributivgesetz** [comment]
What I am trying to say is: We don't need this way of inserting a brain if we have organ manipulation surgery, so why keep it around? It's code duplication.

**Comment by RengaN02** [comment]
> What I am trying to say is: We don't need this way of inserting a brain if we have organ manipulation surgery, so why keep it around? It's code duplication.

If we delete this `attack` code, the changelings will be broken, at the same time we will not be able to revive when we take the brain into another body.

**Pull request closed without merging**